packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -242,6 +262,12 @@ abstract class Artifacts {
     EnvironmentType environmentType,
   });
 
+  /// Retrieve a host specific artifact that does not depend on the
+  /// current build mode or environment.
+  FileSystemEntity getHostArtifact(
```

Comment:
This is essentially the only new API. I'm not 100% about returning a File/Directory instead of the path. Seems like it could be useful.